internal/sveltos/profile.go

```diff
@@ -226,6 +231,91 @@ func GetHelmCharts(ctx context.Context, c client.Client, namespace string, servi
 	return helmCharts, nil
 }
 
+func GetKustomizationRefs(ctx context.Context, c client.Client, namespace string, services []kcm.Service) ([]sveltosv1beta1.KustomizationRef, error) {
+	l := ctrl.LoggerFrom(ctx)
+	kustomizationRefs := []sveltosv1beta1.KustomizationRef{}
+
+	for _, svc := range services {
+		if svc.Disable {
+			l.Info("Skip adding ServiceTemplate", "service_template_name", svc.Template, "is_disabled", svc.Disable)
+			continue
+		}
+
+		tmpl := &kcm.ServiceTemplate{}
+		// Here we can use the same namespace for all services
+		// because if the services slice is part of:
+		// 1. ClusterDeployment: Then the referred template must be in its own namespace.
+		// 2. MultiClusterService: Then the referred template must be in system namespace.
+		tmplRef := client.ObjectKey{Name: svc.Template, Namespace: namespace}
+		if err := c.Get(ctx, tmplRef, tmpl); err != nil {
+			return nil, fmt.Errorf("failed to get ServiceTemplate %s: %w", tmplRef.String(), err)
+		}
+
+		if tmpl.Spec.Kustomize == nil {
+			continue
+		}
+
+		if !tmpl.Status.Valid {
```

Comment:
I don't think we should use `.status.valid` here as it is currently only relevant for helmchart source. Maybe we should update template reconciliation so that the valid field is updated for kustomize and resources sources as well if that's feasible.

Reply:
Yeah the reconciler is setting `.status.valid` within the `updateStatus` function. What I wanted to point out though it that currently `updateStatus` is only being called in relation to helm chart in the [`ReconcileTemplate`](https://github.com/k0rdent/kcm/blob/5f73672be8a5a96f4f3696a50c289120921aa550/internal/controller/template_controller.go#L234) function. So the `.status.valid` field is being set only for helm chart but used here for kustomize (and resources below). Hope that clarifies my concern.

Reply:
I missed reviewing the new servicetemplate_reconciler.go file because github did not show it by default 😄. The `.status.valid` is handled in there so this is good.